src/unix/system.c

```diff
@@ -341,7 +345,27 @@ void Sys_Init(void)
     if (!homedir) {
 	    Sys_Error("Homedir not found!\n");
     }
-    sprintf(homegamedir, "%s/%s", homedir, ".quake2rtx");
+    xdg_data_home_dir = getenv("XDG_DATA_HOME");
+    if (!xdg_data_home_dir) {
+	    xdg_data_home_dir = malloc(sizeof(char) * 
+				       (strlen(homedir) + strlen("/.local/share") + 1));
```

Comment:
```suggestion
				       (strlen(homedir) + strlen("/.local/share") + strlen("\0")));
```
More explicit and no need for a comment.

Reply:
> More explicit and no need for a comment.

This doesn't work as you think. `strlen("\0")` will return 0 (after all, there's nothing but a null terminator in that string).
So better just keep the constant 1.

Reply:
My bad!